src/common/components/ui/formik/validationSchemas.ts

```diff
@@ -22,8 +22,9 @@ import { TFunction } from 'i18next';
 const ALPHANUMBERIC_REGEX = /^[a-zA-Z0-9]+$/;
 const NAME_START_END_REGEX = /^[a-z0-9](.*[a-z0-9])?$/;
 const NAME_CHARS_REGEX = /^[a-z0-9-.]*$/;
-const CLUSTER_NAME_CHARS_REGEX = /^[a-z0-9-]*$/;
-const CLUSTER_NAME_REGEX = /^[a-z0-9](.*[a-z0-9])?$/;
+const CLUSTER_NAME_START_END_REGEX = /^[a-z0-9](.*[a-z0-9])?$/;
+// NOTE: based on https://github.com/openshift/assisted-service/blob/master/internal/cluster/validations/validations.go#L32
+const CLUSTER_NAME_REGEX = /^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)*$/;
```

Comment:
The problem with this regex, while it sure is smarter and perhaps more efficient, the resulting validation messages can get confusing. Like in this case the users will be questioning why is the second validation failing, what is wrong with the characters they used?

![image](https://user-images.githubusercontent.com/87187179/188152929-8b737736-7a6c-4572-9db2-abda6ddc2463.png)

I suggest we keep the conditions just as separate as the messages. If the message is about allowed characters, the validation test shouldn't be looking at anything else. If the message is about the first and last character, again the test should not be looking at what's in the middle.

Reply:
I've merged the two conditions into one to be on the safe side and make the validation compatible with the backend. (https://github.com/openshift-assisted/assisted-ui-lib/pull/1584/commits/6336a16002aef6c54c6133028f3b552c2160f992)
<img width="733" alt="Screenshot 2022-09-20 at 15 22 48" src="https://user-images.githubusercontent.com/1121740/191270260-1217e513-0efa-4197-9646-53df6123c07c.png">

Reply:
I'm still afraid it might confuse some users. The conditions for the cluster name are complicated, but smashing them into one sentence will not help the users.

![image](https://user-images.githubusercontent.com/87187179/191272050-177ad311-343c-4fbf-b1a2-6d341d253871.png)

The conditions for the cluster name are complicated, but smashing them into one sentence will not help the users.

I think we should have something similar for the cluster name as we have for the host names.

![image](https://user-images.githubusercontent.com/87187179/191273407-8c59793b-5a5d-450c-a71b-a414419899f2.png)

Reply:
I understand there are UX concerns, however there is IMHO also a valid concern of a correctness of the separate condition validations. Those are not working correctly at the moment even for the hostnames (putting multiple dots for example) so the individual regexes need to be updated and ensured that they are correct. Otherwise we will hit bugs such as the one I am fixing with this PR. Using the same regex on the UI side as on the backend makes us sure there won't be such bugs.

Anyway, I'll try to craft a regex that resembles the backend one as much as possible in functionality.

Reply:
I've changed the `CLUSTER_NAME_REGEX` to be based off of the backend regex and tweaked it so the validation does not fail for first and last characters which is covered by the separate condition.
<img width="716" alt="Screenshot 2022-09-20 at 17 29 21" src="https://user-images.githubusercontent.com/1121740/191301632-f0d3726f-db29-4d11-af41-3020aaf7a712.png">
<img width="717" alt="Screenshot 2022-09-20 at 17 29 35" src="https://user-images.githubusercontent.com/1121740/191301652-9069fff3-ec79-48ba-812b-994f8eeca655.png">
<img width="718" alt="Screenshot 2022-09-20 at 17 29 58" src="https://user-images.githubusercontent.com/1121740/191301665-dbd4b7bc-5c39-457d-a86f-d604b6ab5608.png">